java/delete-features-feature-service/src/main/java/com/esri/arcgisruntime/samples/deletefeaturesfeatureservice/MainActivity.java

```diff
@@ -0,0 +1,238 @@
+import com.esri.arcgisruntime.mapping.view.MapView;
+
+public class MainActivity extends AppCompatActivity implements ConfirmDeleteFeatureDialog.OnButtonClickedListener {
+
+  private static final String TAG = MainActivity.class.getSimpleName();
+
+  private MapView mMapView;
+
+  private ServiceFeatureTable mFeatureTable;
+
+  private FeatureLayer mFeatureLayer;
+
+  @SuppressLint("ClickableViewAccessibility")
+  @Override protected void onCreate(@Nullable Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_main);
+
+    mMapView = findViewById(R.id.mapView);
+
+    // create a map with streets basemap
+    ArcGISMap map = new ArcGISMap(Basemap.Type.STREETS, 40, -95, 4);
+
+    // create service feature table from URL
+    mFeatureTable = new ServiceFeatureTable(getString(R.string.feature_layer_url));
+
+    // create a feature layer from table
+    mFeatureLayer = new FeatureLayer(mFeatureTable);
+
+    // add the layer to the ArcGISMap
+    map.getOperationalLayers().add(mFeatureLayer);
+
+    mMapView.setOnTouchListener(new DefaultMapViewOnTouchListener(this, mMapView) {
+      @Override public boolean onSingleTapConfirmed(MotionEvent event) {
+        // create a point from where the user clicked
+        android.graphics.Point point = new android.graphics.Point((int) event.getX(), (int) event.getY());
+
+        // create a map point from a point
+        Point mapPoint = mMapView.screenToLocation(point);
+
+        // for a wrapped around map, the point coordinates include the wrapped around value
+        // for a service in projected coordinate system, this wrapped around value has to be normalized
+        Point normalizedMapPoint = (Point) GeometryEngine.normalizeCentralMeridian(mapPoint);
+
+        // identify the clicked feature
+        ListenableFuture<IdentifyLayerResult> results = mMapView.identifyLayerAsync(mFeatureLayer, point, 1, false);
+        results.addDoneListener(() -> {
+          try {
+            IdentifyLayerResult layer = results.get();
+            // get first element found and ensure that it is an instance of Feature before allowing user to delete
+            // using callout
+            if (layer.getElements().size() > 0 && layer.getElements().get(0) instanceof Feature) {
+              inflateCallout(mMapView, layer.getElements().get(0), normalizedMapPoint).show();
+            }
+          } catch (InterruptedException | ExecutionException e) {
+            logToUser(getString(R.string.error_getting_identify_result, e.getCause().getMessage()));
+          }
+        });
+        return super.onSingleTapConfirmed(event);
+      }
+    });
+
+    // set ArcGISMap to be displayed in map view
+    mMapView.setMap(map);
+  }
+
+  /**
+   * Method gets an instance of {@link Callout} from a {@link MapView} and inflates a {@link View} from a layout
+   * to display as the content of the {@link Callout}.
+   *
```

Comment:
Really good method level comment here. I think its a good idea to maybe start taking advantage of javadoc links in future samples.